src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ResetPasswordHandler.php

```diff
@@ -35,7 +35,11 @@ public function __invoke(ResetPassword $command): void
         /** @var ShopUserInterface|null $user */
         $user = $this->userRepository->findOneBy(['passwordResetToken' => $command->resetPasswordToken]);
 
-        Assert::notNull($user, 'No user found with reset token: ' . $command->resetPasswordToken);
+        if (null === $user) {
+            throw new NotFoundHttpException(
```

Comment:
Can you introduce a new custom error and map it to 404 with API platform configuration? This way we won't be coupled to the HTTP namespace of Symfony inside of the handler. I've added an additional phparchitekt rule for that in the future #13913

Reply:
@lchrusciel I introduced `Sylius\Bundle\ApiBundle\Exception\NotFoundExceptionInterface` to avoid adding more and more exceptions to the `exception_to_status` APIP config. Then I created two new very generic exceptions : UserNotFoundException and TokenExpiresException, which now can be used in other API controllers, command handlers, etc...